engine-rest/engine-rest-openapi/src/main/templates/models/org/camunda/bpm/engine/rest/dto/runtime/task/CommentDto.ftl

```diff
@@ -0,0 +1,53 @@
+{
+  "type": "object",
+  "properties": {
+
+    <@lib.property
+        name = "id"
+        type = "string"
+        desc = "The id of the task comment." />
+
+    <@lib.property
+        name = "userId"
+        type = "string"
+        desc = "The id of the user who created the comment." />
+
+    <@lib.property
+        name = "taskId"
+        type = "string"
+        desc = "The id of the task to which the comment belongs." />
+
+    <@lib.property
+        name = "time"
+        type = "string"
+        format = "date-time"
+        desc = "The time when the comment was created." />
+
+    <@lib.property
+        name = "message"
+        type = "string"
+        desc = "The content of the comment." />
+
+    <@lib.property
+        name = "removalTime"
+        type = "string"
+        format = "date-time"
+        desc = "The time after which the comment should be removed by the History Cleanup job.
+                [Default format]($(docsUrl)/reference/rest/overview/date-format/)
+                `yyyy-MM-dd'T'HH:mm:ss.SSSZ`." />
+
+    <@lib.property
+        name = "rootProcessInstanceId"
+        type = "string"
+        desc = "The process instance id of the root process instance that initiated the process
+                containing the task." />
+
+    <@lib.property
+        name = "links"
+        type = "array"
+        dto = "AtomLink"
+        last = true
+        desc = "Link to the newly created task comment with `method`, `href` and `rel`." />
+
+  }
```

Comment:
I added a FTL template in `lib/commons` for the `LinkableDto` since the class is abstract and any properties it defines will be included only in child classes. I made the `description` of the `links` property more uniform, with the only dynamic part being the type of resource(task comment, deployment or process instace, which are the child classes).

Reply:
@mboskamp I changed the implementation so that it uses `allOf`. To do this, I added an extra attribute to the `dto` macro called `extends`. This way we don't have to mix JSON and FTL syntax in `Dto.ftl` files.

The "extended" DTO is referenced at the end of the definition so that its properties are last when seen in the Swagger UI.